desc/objectives/objective_funs.py

```diff
@@ -409,7 +409,7 @@ def _set_things(self, things=None):
         self._things_per_objective_idx = []
         for obj in self.objectives:
             self._things_per_objective_idx.append(
-                [i for i, t in enumerate(unique_) if t in obj.things]
+                [unique_.index(t) for t in obj.things if t in unique_]
```

Comment:
you can get rid of the `if t in unique_` part. In theory `t` should always be in `unique_` unless something goes horribly wrong, and in that case its better to have the `unique_.index(t)` throw an error rather than silently truncating the list